common/forks/forks.go

```diff
@@ -87,3 +88,45 @@ func BlockRange(currentForkHeight uint64, forkHeights []uint64) (from, to uint64
 	}
 	return
 }
+
+// GetHardforkName returns the name of the hardfork active at the given block height and timestamp.
+// It checks the chain configuration to determine which hardfork is active.
+func GetHardforkName(config *params.ChainConfig, blockHeight, blockTimestamp uint64) string {
+	if !config.IsBernoulli(new(big.Int).SetUint64(blockHeight)) {
+		return "homestead"
+	} else if !config.IsCurie(new(big.Int).SetUint64(blockHeight)) {
+		return "bernoulli"
+	} else if !config.IsDarwin(blockTimestamp) {
+		return "curie"
+	} else {
+		return "darwin"
```

Comment:
Can we make fork names an enum instead of string constants?

Reply:
yes. need string constants

Reply:
We could've made them string enums.